Add test for full disk threshold config, leave default as-is

ajfabbri opened the pull request:
## Cover letter

~~Value based on discussion with (internal) customer. We want to default to sane production values, and override where necessary for dev / test. Changed default minimum free disk space threshold to 5 GiB.~~

Fixes #4850 

## Release notes
 * None

Force Push: don't increase default `storage_min_free_bytes`. Instead of "defaulting to prod", default to "dev experience", and handle production defaults at configuration time.

ajfabbri commented:
Force push: fix formatting.

senior7515 commented:
Question. Why not make it a function of partition count with some upper limit. I think this can be pushed to rpk or smth like that. 5g is larger than most dev container setups and will make the getting started experience more complex. I don’t think this belongs in redpanda with a larger limit.

senior7515 commented:
To add more context Testcontainers is a very popular setup for redpanda and is usually super minimal resources so parallelizing those builds will be limited by disk size. Is the 5G gated to non dev mode ? If so I’m ok.

ajfabbri commented:
I followed SRE's lead on this one. @tdewitt.

About next iteration and improvements to design, did you see the Future Work stuff in the RFC?

ajfabbri commented:
I like the idea of leaving the compile-time default smaller, we just need to make sure production users apply better defaults.

ajfabbri commented:
Marking draft. Will probably change this PR to drop the default threshold change, just keeping the config test.

ajfabbri commented:
Filed #5248 for a new rpk subcommand to assist with configuration.

ajfabbri commented:
Ok, ready for review again. Force push: I've removed the configuration default change and updated other issues to set prod. defaults. All that remains in this PR is some python / test tweaks.

emaxerrno commented:
Will review tonight or tomorrow. 

May need pm support too. @mattschumpert

ajfabbri commented:
Will go ahead and merge this since the controversial part of the change (default disk min free bytes config) has been removed. We can always submit a new PR if we come to a different conclusion @tdewitt . This closes #4850, which is needed for coming release.

ajfabbri commented:
CI failure is unrelated.

dotnwat commented:
> CI failure is unrelated.

@ajfabbri link to the issue?
